subscription-engine/kafka-ws/app/Redis/RedisManager.js

```diff
@@ -21,6 +22,22 @@ class RedisManager {
     // TODO: Implement "retry_strategy"
     this.redisClient = redis.createClient(this.config);
     logger.info('RedisManager singleton creation complete!');
+
+    const stateService = 'redis';
+    StateManager.registerService(stateService);
```

Comment:
Within the constructor of the RedisManager, it does not seem the right place to register a service in the StateManager.
Remember that the constructor is used to initialize an instance of the RedisManager object and not the StateManager.

The `healthChecker` method of the `KafkaConsumer` class was cool, use the same approach in `RedisManager` as well:
```javascript
const sigRy = StateManager.signalReady.bind(StateManager, 'redis');
const sigNotRy = StateManager.signalNotReady.bind(StateManager, 'redis');

const obj = new RedisManager();
obj.healthChecker(sigRy, sigNotRy);


// And in the RedisManager class:
class RedisManager {

healthChecker(signalReady, signalNotReady) {
    this.redisClient.on('connect', () => signalReady());
    this.redisClient.on('end', () => signalNotReady());
    this.redisClient.on('reconnecting', () => signalNotReady());
}

```


One more thing, it makes more sense to keep this:
```javascript
StateManager.registerService('redis');
```

inside `StateManager.js`, because that will never change.

Reply:
As you can see this class is a singleton, That's the reason why all StateManager function call concern to this class is in there.